cli.js

```diff
@@ -1,32 +1,30 @@
 #! /usr/bin/env node
+const template = require('./helpers/buildTemplate.js');
+const arguments = require('./helpers/arguments.js');
 
-const fs = require('fs');
-const stampTemplate = require('./helpers/stampTemplate.js');
-const convertTagNameToClassName = require("./helpers/convertTagNameToClassName.js");
-
-// Get the element name from the args
-const elementName = process.argv[2];
-
-if(!elementName) {
-    console.warn("No element name specified!");
-
+// Check the formatting of our arguments
+if(!arguments.ArgumentsHaveElementName()) {
+    console.log("No element name specified");
     process.exit(1);
 }
 
-// if we have an element name, we're good, let's make the template
-const rawTemplate = fs
-    .readFileSync(require.resolve("./templates/singleElement.js"))
-    .toString();
-
-const stamp = stampTemplate(rawTemplate, {
-    className: convertTagNameToClassName(elementName),
-    tagName: elementName
-});
-
-// Now write to file
-const path = process.cwd() + "/" + elementName + ".js";
-
-fs.writeFileSync(path, stamp);
+// Check against the arguments and run the correct action
+switch(process.argv[2].toLowerCase()) {
+    case arguments.MODULE:
+        // Code that is ran if a use passes the --module flag
+        let elementName = process.argv[3];
+        template.CreateElementTemplate(elementName);
+        template.CreateFileTemplate(elementName, "readme.md");
+        template.CreateFileTemplate(elementName, "demo.html");
+        template.CopyFile('.gitignore');
```

Comment:
For all these `template.` things, since this is a common action, can't we just have one method call on template that concatenates all these in some way? Alternately, `CreateFileTemplate` could have syntax like:

```Javascript
template.CreateFileTemplate(elementName, ["readme.md", "demo.html"]);
```